src/code/PreRender.c

```diff
@@ -254,150 +350,339 @@ void func_80170AE0(PreRender* this, Gfx** gfxp, s32 alpha) {
+    //      BackGround = (pMax + pMin) - (ForeGround) * 2
+
+    // OutputColor = cvg * ForeGround + (1.0 - cvg) * BackGround
+    temp = 7 - buffCvg[7];
+    outR = buffR[7] + ((s32)(temp * (pmaxR + pminR - (buffR[7] * 2)) + 4) >> 3);
+    outG = buffG[7] + ((s32)(temp * (pmaxG + pminG - (buffG[7] * 2)) + 4) >> 3);
+    outB = buffB[7] + ((s32)(temp * (pmaxB + pminB - (buffB[7] * 2)) + 4) >> 3);
 
-    pxOut.r = pxR3 >> 3;
-    pxOut.g = pxG3 >> 3;
-    pxOut.b = pxB3 >> 3;
+    pxOut.r = outR >> 3;
+    pxOut.g = outG >> 3;
+    pxOut.b = outB >> 3;
     pxOut.a = 1;
     this->fbufSave[x + y * this->width] = pxOut.rgba;
 }
 
 /**
- * Applies an anti-alias filter to the current prerender
+ * Applies the Video Interface anti-aliasing filter to `this->fbufSave` using `this->cvgSave`
  */
 void PreRender_ApplyAntiAliasingFilter(PreRender* this) {
     s32 x;
     s32 y;
     s32 cvg;
 
+    // Apply AA filter
     for (y = 0; y < this->height; y++) {
         for (x = 0; x < this->width; x++) {
             cvg = this->cvgSave[x + y * this->width];
             cvg >>= 5;
             cvg++;
 
             if (cvg != 8) {
-                PreRender_AntiAliasAlgorithm(this, x, y);
+                // If this pixel has only partial coverage, perform the Video Filter interpolation on it
+                PreRender_AntiAliasFilter(this, x, y);
             }
         }
     }
 }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/PreRender/func_801716C4.s")
+/**
+ * Selects the median value from 9 different 5-bit pixels:
+ * px1[0], px1[1], px1[2], px2[0], px2[1], px2[2], px3[0], px3[1], px3[2]
+ * all args are expected to be an array of 3 different 5-bit values
+ */
+u32 PreRender_Get5bMedian9(u8* px1, u8* px2, u8* px3) {
+    u8 pxValCount[32]; // Stores the count for each of the possible 32 5-bit pixel values
+    u32 pxCount;       // Pixel count
+    s32 pxMed;         // Pixel median value
+
+    // Initialize count to 0 in groups of 4 bits
+    *(s32*)(&pxValCount[0]) = 0;
+    *(s32*)(&pxValCount[4]) = 0;
+    *(s32*)(&pxValCount[8]) = 0;
+    *(s32*)(&pxValCount[12]) = 0;
+    *(s32*)(&pxValCount[16]) = 0;
+    *(s32*)(&pxValCount[20]) = 0;
+    *(s32*)(&pxValCount[24]) = 0;
+    *(s32*)(&pxValCount[28]) = 0;
+
+    // Increment the count that contains the pixel values
+    pxValCount[px1[0]]++;
+    pxValCount[px1[1]]++;
+    pxValCount[px1[2]]++;
+
+    pxValCount[px2[0]]++;
+    pxValCount[px2[1]]++;
+    pxValCount[px2[2]]++;
+
+    pxValCount[px3[0]]++;
+    pxValCount[px3[1]]++;
+    pxValCount[px3[2]]++;
+
+    // Loop through the 32 bits until 5 values are found, then return that bit value.
+    // Note that the median of 9 is the 5th sequential value.
+    pxCount = 0;
+    pxMed = 0;
+    while (true) {
+        pxCount += pxValCount[pxMed];
+        if (pxCount >= 5) {
+            // the median is found
+            break;
+        }
+        pxMed++;
+    }
+
+    return pxMed;
+}
+
+void PreRender_DivotFilter(PreRender* this) {
```

Comment:
We should probably put a comment explaining how this doesn't actually seem like a `DivotFilter`

Reply:
Bump :)